src/utils/wallet.js

```diff
@@ -819,7 +818,7 @@ class Wallet {
             const keyPair = KeyPair.fromString(secretKey)
             await tempKeyStore.setKey(NETWORK_ID, accountId, keyPair)
             account.keyStore = tempKeyStore
-            account.inMemorySigner = new nearApiJs.InMemorySigner(tempKeyStore)
+            account.inMemorySigner = account.connection.signer = new nearApiJs.InMemorySigner(tempKeyStore)
```

Comment:
`account.connection.signer = `

why is this needed? This doesn't look good to me. Why not create temporary `Account` instance  with necessary config instead of patching existing one?

Reply:
was always confused by this:
https://github.com/near/near-wallet/blob/2fea47b429f844af95a4b21c60a0e081369575e1/src/utils/wallet.js#L629

Reply:
looks like it should have been `this.inMemorySigner` (field in `Wallet` object)